src/osd/OSD.h

```diff
@@ -1137,26 +1138,51 @@ class OSDService {
 
   // -- OSD Full Status --
 private:
+  friend TestOpsSocketHook;
   Mutex full_status_lock;
-  enum s_names { NONE, NEARFULL, FULL, FAILSAFE } cur_state;  // ascending
+  enum s_names { INVALID = -1, NONE, NEARFULL, BACKFILLFULL, FULL, FAILSAFE } cur_state;  // ascending
```

Comment:
can we take this chance to make `full_status_lock` a mutable member variable? so we can mark the getter methods using it `const`?